tests/Rules/CheckTypehintCallerTypeRule/Fixture/SkipGenericType.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Tests\Rules\CheckTypehintCallerTypeRule\Fixture;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+
+/**
+ * @template T of MethodCall
+ */
+class SkipGenericType
+{
+    /**
+     * @param T $node
+     * @return void
+     */
+    public function run(Node $node)
+    {
+        $this->getsAGeneric($node);
+    }
+
+    /**
+     * @param T $node
+     * @return void
+     */
+    private function getsAGeneric(Node $node)
```

Comment:
before this PR the rule reported

> Parameter 1 should use "T of PhpParser\Node\Expr\MethodCall" type as the only type passed to this method

which doesn't make sense because php-src doesn't support generics natively